langkit/templates/python_api/iterator_types_py.mako

```diff
@@ -0,0 +1,87 @@
+## vim: filetype=makopython
+
+<%def name="base_decl()">
+
+class _BaseIterator(object):
+    """
+    Base class for Ada iterator bindings.
+    """
+
+    c_element_type = None
+    """
+    Ctype class for iterator elements.
+    """
+
+    __slots__ = ('c_value',)
+
+    def __init__(self, c_value):
+        self.c_value = c_value
+
+    def __repr__(self):
+        return '<{}>'.format(type(self).__name__)
+
+    def clear(self):
+        self.c_value = None
+
+    def __del__(self):
+        self.dec_ref(self.c_value)
+        self.clear()
+
+    @classmethod
+    def wrap(cls, c_value):
+        res = cls(c_value)
+        # inc_ref?
+        return res
+
+</%def>
+
+<%def name="decl(cls)">
+
+<%
+    element_type = cls.element_type
+    c_element_type = pyapi.c_type(element_type)
+%>
+
+class ${cls.py_converter}(_BaseIterator):
+    """
+    Wrapper class for iterators of ${cls.element_type.name}.
```

Comment:
This documentation is for users, so I suggest just writing: `Iterator over ${cls.…}`